hikari/impl/shard.py

```diff
@@ -177,53 +177,60 @@ async def send_json(
         await self.send_str(pl, compress)
 
     async def _receive_and_check(self, timeout: typing.Optional[float], /) -> str:
-        buff = bytearray()
+        message = await self.receive(timeout)
 
-        while True:
-            message = await self.receive(timeout)
+        if message.type == aiohttp.WSMsgType.TEXT:
+            return message.data  # type: ignore
```

Comment:
why not assert the type?

Reply:
This was already like this, I only moved code around. Can be changed tho

Reply:
Yeah I'd appreciate changing it even if its not really the purpose of this pr 👍

Reply:
Added